teuthology/openstack/__init__.py

```diff
@@ -804,9 +804,15 @@ def run_suite(self):
             else:
                 argv.append(original_argv.pop(0))
         if self.args.test_repo:
-            repos = [{'name':k, 'url': v}
-                                    for k, v in [x.split(':', 1)
-                                    for x in self.args.test_repo]]
+            def repo(name, url):
+                if '!' in name:
+                    n, p = name.split('!', 1)
+                    return {'name': n, 'priority': p, 'url': url}
```

Comment:
This comes out like so:

```
      - name: internal
        priority: '98'
        url: http://10.86.0.120/artifacts/ci/bcebbc4f96c438f3ef61c7657ec11b4585f7299f4ec75a59adee2bce79303d2f/SUSE-Enterprise-Storage-6-POOL-Internal-x86_64-Build3.8
```

Please use `int(p)` instead of bare `p`.

Reply:
why it's required? I would like to left it as string

Reply:
everywhere else, we take care not to have numbers as strings in the yaml dump

Reply:
what if I want in future to have priority by reference to another repo?

Reply:
I would argue that is a bad idea. Priority should always be an integer.